src/compiler/checker.ts

```diff
@@ -6607,12 +6606,18 @@ namespace ts {
                     }
                 }
                 if (isBindingPattern(declaration.name)) {
-                    return getTypeFromBindingPattern(<BindingPattern>declaration.name);
+                    return createImpliedType(getTypeFromBindingPattern(<BindingPattern>declaration.name));

```

Comment:
So this step applies both for top level initializers as well as initializers nested in binding patterns?